Add rover ackermann model

chfriedrich98 opened the pull request:
Adds a gz-garden model for an ackermann rover that is to be used by the new ackermann module (https://github.com/PX4/PX4-Autopilot/pull/23024). 
Also adds a new world specifically for rovers.

PerFrivik commented:
@Jaeyoung-Lim if you have time :D

Jaeyoung-Lim reviewed (changes requested):
Sorry for the late reply, I was away.

Could you simplify the mesh files? They are too big (one of them being 48MB) to exist in a git tree. It would be great if they are less than 1MB

Jaeyoung-Lim commented on worlds/default.sdf:
```diff
@@ -2,9 +2,9 @@
+      <max_step_size>0.002</max_step_size>
       <real_time_factor>1.0</real_time_factor>
-      <real_time_update_rate>250</real_time_update_rate>
+      <real_time_update_rate>500</real_time_update_rate>
```
Can you elaborate on how it segfaults and why? Does it segfault on the gz side? or the px4 instance?

PerFrivik replied:
I tried a lot of things and in the end, this is what worked, it segfaults on the GZ side, I had a similar issue when I was working on a boat sim, and increasing the update rate worked well.

PerFrivik replied:
I think the reason this happens is because of the steering, when we change the steering angle, part of the wheel goes into the ground, and I believe if the step is too large, it gives room for the force calculation to diverge and explode.

Jaeyoung-Lim replied:
Not sure if I understand. Why does this not happen to a drone standing still or colliding with the ground?

Jaeyoung-Lim commented on models/rover_ackermann/model.sdf:
```diff
@@ -0,0 +1,500 @@
+		<i_gain>0</i_gain>
+		<d_gain>0</d_gain>
+	</plugin>
+	  <!-- <plugin
+	    filename="ignition-gazebo-ackermann-steering-system"
+	    name="ignition::gazebo::systems::AckermannSteering">
+	    <left_joint>rover_ackermann/FrontLeftWheelJoint</left_joint>
+	    <left_joint>rover_ackermann/RearLeftWheelJoint</left_joint>
+	    <right_joint>rover_ackermann/FrontRightWheelJoint</right_joint>
+	    <right_joint>rover_ackermann/FrontRightWheelJoint</right_joint>
+	    <left_steering_joint>servo_0 </left_steering_joint>
+	    <right_steering_joint>servo_1</right_steering_joint>
+	    <kingpin_width>.18</kingpin_width>
+	    <steering_limit>0.5</steering_limit>
+	    <wheel_base>.2255</wheel_base>
+	    <wheel_separation>.2</wheel_separation>
+	    <wheel_radius>0.0365</wheel_radius>
+	    <min_velocity>-100</min_velocity>
+	    <max_velocity>100</max_velocity>
+	    <min_acceleration>-5</min_acceleration>
+	    <max_acceleration>5</max_acceleration>
+	    <topic>/cmd_vel</topic>
+	  </plugin> -->
```
```suggestion
```

Jaeyoung-Lim commented on models/rover_ackermann/model.config:
```diff
@@ -0,0 +1,11 @@
+    <name>Benjamin Perseghetti</name>
+    <email>bperseghetti@rudislabs.com</email>
+  </author>
+  <description>A model of the NXP Mobile Robotics Buggy3.</description>
```
Where is this model copied from? Please confirm that the license for this model permits us to have this model under our repository?

PerFrivik replied:
I got this model from the following [PR](https://github.com/PX4/PX4-Autopilot/pull/21273) opened by Alex and I asked him if we could use it in Discord a while back and he agreed.

Jaeyoung-Lim replied:
@PerFrivik But Alex is not the author of this model?

chfriedrich98 commented:
@Jaeyoung-Lim Sorry for the very long delay, there were a couple of things I had to clear up first.
I received permission from @bperseghetti to use his model as the base of this one if I include the license which I added in https://github.com/PX4/PX4-gazebo-models/commit/8ff172aad25dfe47f89ba5409ec69b8c5e11e2a3 (It is the same license that is included with the x500 models).
In regards to simplifying the mesh files: I switched the originial ones with my own, which are  below the 1MB you mentioned.
Relating to the segfault: If we don't want to change the default world, I could add a world specific for rovers. This would be nice anyway, because it would be very useful to have some pattern on the floor to give some points of reference.

Jaeyoung-Lim reviewed (changes requested):
@chfriedrich98 Thanks for addressing the reviews.

If you can clarify why and what is the problem with the default world, we can also change it. 
Otherwise could you add a custom world for the rover in that case? I am just not comfortable of changing the default world without understanding what is actually the issue and implications of the change.

chfriedrich98 commented:
@Jaeyoung-Lim I removed the changes I made to the default world and added a rover specific world that is based on the default one with some slight changes (added grid for reference, increased update frequency/smalled timesteps and increased friction).

Claudio-Chies commented:
@chfriedrich98 could you also add the rover world to the gz_bridge [CMakeLists](https://github.com/PX4/PX4-Autopilot/blob/main/src/modules/simulation/gz_bridge/CMakeLists.txt)? this way you are able to use the world with `make px4_sitl gz_rover_ackermann_rover`

hamishwillee commented:
@chfriedrich98 Can we have an update to:
- https://docs.px4.io/main/en/sim_gazebo_gz/#running-the-simulation
- https://docs.px4.io/main/en/sim_gazebo_gz/vehicles.html

Also for you and @Claudio-Chies - can you please help with https://github.com/PX4/PX4-user_guide/pull/3338

I'm trying to mirror the gazebo-classic repo to make it easy to understand the world options, what they do, when they make sense to use, etc. Associated with this PR, for example, you might add a note in the world that is used when you command `make px4_sitl gz_rover_ackermann_rover`

chfriedrich98 commented:
@hamishwillee Basic documentation for the rover simulation is included in https://github.com/PX4/PX4-user_guide/pull/3311.
I'll add more info on the rover world there aswell.
